### ISSUE TITLE
Extending wing flaps at too high airspeed should get them stuck / rip them off

### ISSUE THREAD

**Issue by hbeni** [HHS81/c182s#463]
May be simulated easily with:
- [x] getting stuck -> triggering `serviceable`
- [x] ripping off -> 
  - [x] hide model (also for MP)
  - [x] add negative lift coefficient to simulate reduced lift
- [x] Maybe we then can get rid of the message from the simple limits framework
- [x] Make repairable

POH 2-4/2-5:
![grafik](https://user-images.githubusercontent.com/13608602/146274800-769c3c8f-1ae6-40d7-b1f2-01ac4c9410b7.png)
![grafik](https://user-images.githubusercontent.com/13608602/146274874-016fb406-bb8b-4864-abec-1a335656b074.png)
POH 7-13 describes the wing flap system in detail.
![grafik](https://user-images.githubusercontent.com/13608602/146677508-c26adfef-621e-4c14-a0a5-5baa7eb976d7.png)

----
- May be also done together with #464
- some reports on consequences: https://www.pprune.org/pacific-general-aviation-questions/260242-flaps-question-you-cessna-fanatics.html

**Comment by wlbragg** [comment]
I really like this one. I always just throw some flaps for various reason without regard to speed. I know it is wrong and I should have to pay the price. This will be a nice addition.

**Comment by hbeni** [comment]
Ohhh you will pay for sure :)
Do you have real world data to back my code up?
Currently I „invented“ some numbers based on the POH (limits + some leeway), but for instance I do not know how severe ripped off flaps are for lift (currently i shoot for -15% per lost flap).
And another thing is that currently i just can damage both flaps the same time - i have no forces data for distinguishing them. Does the c172 wing damage code supply lift data for each wing individually? (Maybe we can derive something off of that, maybe a factor to apply to each flap-break-limit)

**Comment by hbeni** [comment]
@wlbragg You ca test now :)
https://github.com/HHS81/c182s/pull/467

**Comment by ysopflying** [comment]
A few thoughts:
Aircrafts should be fail safe. A sensible design would make the actuator the weakest link, meaning the flap would rather retract than the whole things being ripped off. Basic engineering, it should also be reflected in the regulations somewhere.

Left/right forces are usually not simulated. As also maintenance, exact material specification and bad luck have a random influence, a randomized factor could determine, which flap is damages first.

Another possibility is to let the one damage first the aircraft rolls into due to higher local alpha and resulting higher loads.

As end of white arc is maximum for 30° flap, it is possible to calculate a maximum possible deflection at higher speeds to trigger the damage. More on this later, if you like.

**Comment by hbeni** [comment]
> Aircrafts should be fail safe. A sensible design would make the actuator the weakest link, meaning the flap would rather retract than the whole things being ripped off. Basic engineering, it should also be reflected in the regulations somewhere.

This is implemented. Currently the damage system has two thresholds, one for "getting stuck" and a higher one for "ripping off". See here for the actual thresholds currently used:
https://github.com/HHS81/c182s/blob/c67a6acfd5368530b868c79bc21926cfc38fbfa2/Systems/damage.xml#L184-L211

> Left/right forces are usually not simulated. As also maintenance, exact material specification and bad luck have a random influence, a randomized factor could determine, which flap is damages first.

Nice idea. I think that could be copied over from the sparkPlugFouling implementation, as we also have such a factor there.
OTOH what we mostly likely see will be both flaps getting stuck if one flap stucks, because they are linked, if I remember correctly? So in this case it won't matter which flap comes first, there will be no visible cue to the pilot, and we may save the (small) performance.

> Another possibility is to let the one damage first the aircraft rolls into due to higher local alpha and resulting higher loads.

This could be added by calculating additional force depending on the roll rate and direction. As far as I have seen, the wing damage model does this, however I'm not so sure how to implement that. An quick&dirty way would be just another table that uses the already calculated data from the wing load table, and translate that into some "additional reduction of speed limit". But how much?

> As end of white arc is maximum for 30° flap, it is possible to calculate a maximum possible deflection at higher speeds to trigger the damage. More on this later, if you like.

Already implemented, see table of first quote answer. The limits are caluclated by the actual flap degree set, so moving them also matters.

**Comment by wlbragg** [comment]
No, no data. I suspect it is likely more than we would think, poh limits. and then some. But I don't know that.

As far as percentage, do you know how much lift we get when deployed to full?

**Comment by ysopflying** [comment]
Summarizing a few lines of hand calculation:
A flap is hinged on two bolts. In case the actuator failed completely, the flap will retract and fly with the aircraft. Detachment only, if the aircraft disintegrates out of other reasons.
If the actuator is stuck and holds the flap in position, the flap is connected with three load carrying points.
Using a conservative approach and with only two of them effective, assuming probable dimensions and material, nothing will happen below 180kts with 90° deflection. Actual limits will be higher. Of course this can be refined by applying exact area and lift/drag coefficients, but the outcome is, that a properly maintained flap will stay with the aircraft. 
Would be great to find that backed up in the regulations, Should be in it. 

Between 10° and 0° it could be linear to v², so you could add a 5° limit at 1.41 times 10° limit.

**Comment by hbeni** [comment]
Well but this also means we don’t need to merge this pr because damage to the flaps are actually way out of limits? Or did i understood dorthin wrong here?

> Between 10° and 0° it could be linear to v², so you could add a 5° limit at 1.41 times 10° limit.

what happens after the limit?

**Comment by hbeni** [comment]
Or do we model a stuck flap as soft damage (as now), and instead of separation at high we just break the actuator (ie flaps=0)?

**Comment by ysopflying** [comment]
> Well but this also means we don’t need to merge this pr because damage to the flaps are actually way out of limits? Or did i understood dorthin wrong here?
> 
Stuck flaps due to overspeed sounds like a great addition.
Loosing flaps is not realistic. Unless it could be possible in the preflight check, that nuts and bolts are missing...


> > Between 10° and 0° it could be linear to v², so you could add a 5° limit at 1.41 times 10° limit.
> 
> what happens after the limit?

Pretty much nothing I suppose. You'll have number of other problems with your aircraft, if you are so fast, that 3° of flap deflection make an overload situation. So you could just input a line for the 5°.

**Comment by ysopflying** [comment]
> Or do we model a stuck flap as soft damage (as now), and instead of separation at high we just break the actuator (ie flaps=0)?

Sounds cool! At least it is one of the possibilities. It would need a deeper look into the mechanics, how it could bend or break.

**Comment by hbeni** [comment]
More stuff to read:
- https://www.pilotsofamerica.com/community/threads/flap-overspeed-inspection.127959/
- https://mooneyspace.com/topic/35464-flap-overspeed/
- https://www.airliners.net/forum/viewtopic.php?t=746531
- https://vansairforce.net/community/showthread.php?s=f1a20ae1b4aed4975976cda9e78a718d&t=147242&page=2
- https://vansairforce.net/community/showthread.php?t=158061&page=2
- https://commons.erau.edu/cgi/viewcontent.cgi?article=1026&context=jaaer
-

**Comment by hbeni** [comment]
Seems the most common issue is a bent aluminium strut rod. But most overlimit reports above are noting that no harm was done.

Tl;dr: I’m not sure we should merge this. Maybe it’s better to just leave the current message on the screen…

**Comment by ysopflying** [comment]
To give a recommendation considering we are doing serious simulation here:
- either finding C182-specific damage reports
- calculating through the system (all dimensions, all material specifications needed) and find the weakest link with its failure mode.
With a bit of guessing and reading the links above it could be, that buckling of the pushing rods is limiting. If they are bent heavily the system could block.

To come to a conclusion:
If flaps extended above the limits, they are to be inspected (just in case you plan maintenance properties :) )
If these limits are exceeded by 30% the flaps either stuck or retract (50-50 randomized)
Of course the last part is speculation but might be not that far off and gives the pilot an experience to remember.

**Comment by hbeni** [comment]
Sounds good to me.

**Comment by ysopflying** [comment]
if they bend, they will be stuck in position. As the rods are being pushed, they would rather buckle than failing completely. This would also take the loads off.
If flaps retract immediately, this could happen because the locking mechanism fails (there must be some kind of brake, otherwise the electric motors have to act against the forces all the time.)
Difficult to judge what really happens. That's why the hands-on approach with randomization. 

It should be possible to model both flaps separately. On the sketch in the POH they are connected by pulleys. This could of course also fail somehow, maybe on the side the aircraft is rolling into. Not sure, if this is a typical damage...

**Comment by hbeni** [comment]
> must be some kind of brake

Not actually: afaik it works by a motor rotating a threaded rod, thereby „screwing it out“ when extending. When this bends then no screwing is possible anymore, and also the motor does not need to hold the lead itself, that is done by the threads.

**Comment by ysopflying** [comment]
ah ok. "Friction brake" :)

**Comment by hbeni** [comment]
Looking at the diagram at POH 7-13, it looks like this for me:
- the electrical motor sits behind the firewall
- based on the flap lever switch setting, it rotates, thereby pulling a cord
- that cord pulls via pulleys o a pull/push rod
- the rods are mechanically linked, so they move together.

For simulation purposes this means:
- that they will get stuck together
- Breakage of the rod on one flap will make that flap move to 0°, BUT as found out above, this should be a very rare thing to meet. Like I did it now (break = VFe+50%) the only value below Vne is the full flap setting. And try to reach that value with full flaps! :)

I currently prepare another branch implementing just the stuck/rod-breakage thingy. And I try to get the roll moment into the equation, so the roll actually puts some more load onto the flap (which means both get stuck more quickly and breakage can occur also maybe be encountered the 20° setting).

**Comment by ysopflying** [comment]
Nice research -> good approach!

Correct, instantaneous will only occur, if the connection of the rod to flap fails or the threads are being torn of. The scenario, that they get stuck on one position seems to much more likely.

**Comment by hbeni** [comment]
Again, some reading about breakage.
Currently, we assume (rightfully) that a broken flap should make it retract to 0° instantly due do air pressure.
Two questions come to my mind:
- Is this probable enough that we should simulate it (after all, there is a (really) small fps cost to this!)?
- If we want to simulate it - can a flap go to 0° while the other stays deployed?


Some links I dug out this evening:
- https://www.pilotsofamerica.com/community/threads/c172-asymmetrical-flap-deployment.73971/
- https://www.pprune.org/private-flying/620041-c172-flap-asymmetry.html
- https://www.aopa.org/news-and-media/all-news/2000/march/flight-training-magazine/applying-flaps-in-a-turn--good-or-bad-idea ("Here's an excerpt from Aviation Safety magazine (January 1st issue, 1986) that speaks to split flap conditions:
A Cessna spokesman said most Cessna single-engine aircraft have undergone split-flap testing, **even though the typical Cessna cable interconnect is considered to fall under the failure-proof part of the rule [FAR Part 23]**. According to Cessna, **a split-flap condition in flight "will not produce a great deal of change**. Normal pilot responses in roll and yaw can control the aircraft from Vfe [maximum flap extended speed] down to the stall speed. ")
- https://www.tsb.gc.ca/eng/rapports-reports/aviation/2017/a17o0209/a17o0209.html


**Conclusion:**
- I really start to think, that we should just model the stuck damage.
- The only real possible damage could be a breakage of the linkage of both flaps, which means that one can be actuated, and the other not anymore (only one side stuck). Causes for that rely in bad maintenance and corrosion (wear and tear we don't simulate yet).
- The forces required to _break_ the push rod (or the threads) seem to be so high that it only occurs due to crash landings and not in flight due flying way over limits.
- Even _if_ a split flap condition occured, it seems that the plane handling does not change that drastically, and therefore we really might choose to drop the idea of simulating this rare thing.


So I'm inclined to simplifying the code a great deal (just set single limits on stuck for both flaps, to save 2 fdm calculations) and not simulate the breakage at all. The only bad thing, flying way overspeed, will be stucked flaps then.

**Comment by ysopflying** [comment]
Sounds good, as it adds the most probable damage scenario to the user experience and gives him feedback, where to improve his flying style.

In case asymmetric situations are being modelled as part of a (non-)maintenance module, the additional moments and forces can be easily calculated.

**Comment by hbeni** [comment]
OK, did it, see the _new_ PR/branch Issue463-FlapsDamage-2

Ready for testing

**Comment by wlbragg** [comment]
Wow, step away for a day and see what happens. Yeah, sounds very complicated. One's first thoughts on the subject don't totally hold up under scrutiny. It would be nice to have a list of actual events of "like aircraft" and see all the manifestations of this potential damage.
One other thought, does the random failure feature figure in here any better?

**Comment by hbeni** [comment]
> One other thought, does the random failure feature figure in here any better?

Basicly its the same outcome, however the underlaying cause is different. The failure mode simulates failure of the flap motor.

**Comment by ysopflying** [comment]
>  It would be nice to have a list of actual events of "like aircraft" and see all the manifestations of this potential damage.?

Would be great to have such a list including the limits. But coming up with such a list is not easy. Well, of course it is possible to calculate the structure and find the weak points. But for the aircrafts we are building we do not have access to structural plans.
So questions like for example, if wing or horizontal stabilizers flutter first, or if they fail due to divergence requires deeper research.
Sometimes damage reports are known. 
I don't want to be guilty of self-citing so close before Christmas, but there was a forum post on this.

Maintenance failures would be fun to simulate as well. Whole new story.
Years of coding fun :)

**Comment by hbeni** [comment]
Well, the essential limis are reported in the poh, and i guess it’s our most important task to have side feedback after exceeding them, so the pilot has a proper sense on what he did wrong.

**Comment by ysopflying** [comment]
Absolutely. Limits are in the POH and it is nice to give the pilot feedback.
So we know, WHEN things are happening, but not WHAT is happening.

But maybe we dive into it too much. Main point is, that our nice aircrafts should be handed with care.

**Comment by hbeni** [comment]
So, I think we now just need some more flight testing, to verify the PR `Issue463-FlapsDamage-2 ` does not break things.
Then we can merge.

**Comment by ysopflying** [comment]
lunch break testing on 2020.3.8. Worked fine.
Neat feature to have the flaps in some kind of "in between" setting as shown on the indicator.
Suggest to lower the limits and use sqrt(1.5) to be in line with the general safety factor instead of +30%.

**Comment by hbeni** [comment]
What values do you suggest exactly for my table?

**Comment by ysopflying** [comment]
0   999
                            10   175
                            20   147
                            38   122

**Comment by hbeni** [comment]
Done :) Pushed and ready for testing

**Comment by ysopflying** [comment]
Tested. Great work!
